packages/components/src/Modal/Modal.rebuilt.test.tsx

```diff
@@ -76,41 +76,153 @@ describe("Composable Modal", () => {
-    render(
-      <Modal.Provider open={true}>
-        <Modal.Content>
-          <Modal.Header title="Billing Settings" />
-        </Modal.Content>
-      </Modal.Provider>,
-    );
+  describe("accessibility", () => {
+    it('modal contains aria role of "dialog"', async () => {
+      render(
+        <Modal.Provider open={true}>
+          <Modal.Content>
+            <Modal.Header title="Modal Title" />
+          </Modal.Content>
+        </Modal.Provider>,
+      );
+      expect(await screen.findByRole("dialog")).toBeInTheDocument();
+    });
 
-    expect(
-      await screen.findByRole("dialog", { name: "Billing Settings" }),
-    ).toBeInTheDocument();
-  });
+    describe("dialog is named from title prop", () => {
+      it("when ariaLabel is not provided", async () => {
+        render(
+          <Modal.Provider open={true}>
+            <Modal.Content>
+              <Modal.Header title="Billing Settings" />
+            </Modal.Content>
+          </Modal.Provider>,
+        );
 
-  it("uses ariaLabel via aria-label only when no title/header is provided", async () => {
-    render(
-      <Modal.Provider open={true} ariaLabel="Add Customer">
-        <Modal.Content>{/* no Modal.Header on purpose */}</Modal.Content>
-      </Modal.Provider>,
-    );
+        const dialog = await screen.findByRole("dialog");
+
+        expect(
+          screen.getByRole("dialog", { name: "Billing Settings" }),
+        ).toBeInTheDocument();
+        expect(dialog).toHaveAttribute("aria-labelledby");
+      });
+
+      it("when ariaLabel is provided (title takes precedence)", async () => {
+        render(
+          <Modal.Provider open={true} ariaLabel="This should be ignored">
+            <Modal.Content>
+              <Modal.Header title="Billing Settings" />
+            </Modal.Content>
+          </Modal.Provider>,
+        );
+
+        const dialog = await screen.findByRole("dialog");
+
+        expect(dialog).toBeInTheDocument();
+        expect(
+          screen.getByRole("dialog", { name: "Billing Settings" }),
+        ).toBeInTheDocument();
+        expect(dialog).toHaveAttribute("aria-labelledby");
+      });
+    });
+
+    describe("dialog is named from ariaLabel", () => {
+      it("when no header is provided", async () => {
+        render(
+          <Modal.Provider open={true} ariaLabel="Add Customer">
+            <Modal.Content>{/* no Modal.Header on purpose */}</Modal.Content>
+          </Modal.Provider>,
+        );
 
-    expect(
-      await screen.findByRole("dialog", { name: "Add Customer" }),
-    ).toBeInTheDocument();
+        const dialog = await screen.findByRole("dialog");
+
+        expect(
+          screen.getByRole("dialog", { name: "Add Customer" }),
+        ).toBeInTheDocument();
+        expect(dialog).toHaveAttribute("aria-label", "Add Customer");
+      });
+
+      it("when header lacks id/modalLabelledBy", async () => {
```

Comment:
our "test" or "it" should be a helpful description of the behavior we are asserting. that way when a test fails it tells the reader something about what went wrong

ie. "it should be identifiable by ariaLabel"

"when header lacks id/modalLabelledBy" only tells me the conditions, not what to expect. I think this would be better suited for a "describe"